pkg/http/routes.go

```diff
@@ -0,0 +1,32 @@
+package api
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+)
+
+const (
+	// DefaultPort is the Default setting a port is set to for the server
+	// when not otherwise specified. It is publicly available for other
+	// clients to lookup, and potentially use.
+	DefaultPort = 8081
+)
+
+type StartOptions struct {
+	Port     int
+	BindHost string
+}
+
+// Start runs the cragcast API server.
+func Start(opts StartOptions) {
+	controller := New()
+	http.HandleFunc("/forecast/boulder", controller.GetForecast)
+
+	if opts.Port == 0 {
+		opts.Port = DefaultPort
+	}
+	address := fmt.Sprintf("%s:%d", opts.BindHost, opts.Port)
+	log.Printf("serving cragcast API at %s", address)
+	log.Fatal(http.ListenAndServe(address, nil))
```

Comment:
Possibly...I'm not sure I'm clear on what you mean by application entry? Can you describe where you'd _ideally_ think it should sit?

Reply:
application entry is basically `cmd/main.go` in my mind. It's all the shit that happens when you start up the application.

Right now you've got some infra set up to handle CLI commands; however,  there isn't a real need for CLI entry as this is just a simple web server so instantiation of the server could just be part of a StartServer function in cmd/main.go. With all that being said, I'm sure there's a use case for some CLI commands so putting the `http.ListenAndServe` logic outside of `main.go` is fine but it feels weird to put it as part of the routing layer.

Reply:
Ah I see.

Keep in mind that what we produce **will be** a command line tool (static binary), even though the main use case is to server APIs via HTTPS. Unlike node, we have no runtime, so `cragcast` will be a binary. By giving it more formal CLI-qualities, it makes it easier to pump options into it that inform how the server should run. This could be provided as flags and/or config files to the end state binary.

Does that help reason about it? Promise this isn't just an "oh cool here's a CLI" thing but more so a long-term model that we'll use for literally running the server.